src/widgets/lv_switch.h

```diff
@@ -32,8 +27,15 @@ extern "C" {
  *      TYPEDEFS
  **********************/
 
+typedef struct {
+    lv_obj_t * sw;
+    int32_t anim_state;
+} _lv_switch_anim_t;
```

Comment:
Do we really need this `struct`? The switch can have on one knob so it seems we can store `anim_state` in `lv_switch_t`.